config/backend-config/backend-config.go

```diff
@@ -262,6 +318,11 @@ func (bc *backendConfigImpl) StartWithIDs(ctx context.Context, workspaces string
 	bc.ctx = ctx
 	bc.cancel = cancel
 	bc.blockChan = make(chan struct{})
+	bc.cache = cacheOverride
+	if bc.cache == nil {
+		cacheKey := fmt.Sprintf(`%x`, sha1.Sum([]byte(workspaces))) // using a fixed size key	// skipcq: GSC-G401, GO-S1025
```

Comment:
Why not use a sha256 ?

Reply:
we don't really care about security here, what we really care about is to have a cache key of fixed size, since the list of workspaces can be pretty big. Think about git commit hashes, they are sha1 too and it is ok

Reply:
I understand, still using 256 would not hurt. btw git is moving to sha256.  

I would suggest against using multiple `workspaceIDs` and thus the need for hashing. 

Given how fast the set of workspaceIDs might change in some environments, we would end up with a lot of unneeded entries, plus increasing the chances of downtime in-between workspaceID changes.

I would propose using: `{deploymentType}/{Identifier}`
* `dedicated/{workspaceID}`
* `namespace/{namespaceName}`
 
This way our key will be the same as long as environment configurations do not change. Some workspaces might be missing if downtime happens before the new workspaceID is fetched but thus is better than not having any configs.

Reply:
>we would end up with a lot of unneeded entries

every time `workspaces` changes, the previous entries are cleared.
>This way our key will be the same

And if we want to make the key immutable for a server's lifetime, we don't even need one imo. Just one column(`config`) would suffice for this purpose. We could avoid using `sha#`  entirely.

Reply:
we will be using the Identifier.ID as a cache key (workspaceID for single-tenant/namespace for multi-tenant)